### PR TITLE
Rename trashcan api calls for open/close lid.

### LINKED ISSUE #4061: Trashcan api cleanup
Currently the trashcan emits an event when the flyout opens and closes `"trashcanOpen".` There also exists an `open`/`close` method on the trashcan which is not related to the flyout opening and closing (but rather the lid opening and closing).
This inconsistency in terminology makes it unclear what it means for "trashcan to be open".

**Describe the solution you'd like**
Review and rename api calls (particularly `open` and `close`) so that it is clear when a method is animation the lid vs opening the flyout. In this scenario, ideally nothing would say "open the trashcan" and would rather either say "open the trascan lid" or "open the trashcan flyout"

**Describe alternatives you've considered**

Let opening the trashcan mean opening the flyout (and update terminology for opening the lid) since opening the flyout is moreso the main function of the trashcan (rather than the lid animation).

**Additional context**
#4024 introduced the new UI Event "trashcanOpen"

### PR DESCRIPTION
**Pull request by moniika**
## The basics

- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Fixes #4061

### Proposed Changes

Renames methods for opening/closing trashcan lid to mention "lid".

### Reason for Changes

This rename makes it clearer that these methods are referring to closing the lid, rather than the trashcan flyout (which can also be opened/closed).

### Test Coverage

Tested on:
 * Desktop Chrome 

### Additional Information

I searched devsite source and didn't find any places that should be updated, aside from the pages auto-generated from jsdoc.

### CODE REVIEW

File: core/bubble_dragger.js [hunk 1]
```diff
@@ -145,14 +145,14 @@ Blockly.BubbleDragger.prototype.maybeDeleteBubble_ = function() {
...
   if (this.wouldDeleteBubble_) {
     if (trashcan) {
-      setTimeout(trashcan.close.bind(trashcan), 100);
+      setTimeout(trashcan.closeLid.bind(trashcan), 100);
     }
     // Fire a move event, so we know where to go back to for an undo.
     this.fireMoveEvent_();
     this.draggingBubble_.dispose(false, true);
   } else if (trashcan) {
     // Make sure the trash can is closed.
```
Comment (alschmiedt) [inline core/bubble_dragger.js:154]: nit: trash can lid is closed.

File: core/block_dragger.js [hunk 2]
```diff
@@ -287,15 +287,15 @@ Blockly.BlockDragger.prototype.maybeDeleteBlock_ = function() {
...
     if (trashcan) {
-      setTimeout(trashcan.close.bind(trashcan), 100);
+      setTimeout(trashcan.closeLid.bind(trashcan), 100);
     }
     // Fire a move event, so we know where to go back to for an undo.
     this.fireMoveEvent_();
     this.draggingBlock_.dispose(false, true);
     Blockly.draggingConnections = [];
   } else if (trashcan) {
     // Make sure the trash can is closed.
```
Comment (alschmiedt) [inline core/block_dragger.js:297]: nit: lid is closed.

File: core/bubble_dragger.js (hunk 1 above)
Comment (moniika) [reply core/bubble_dragger.js]: Updated

File: core/block_dragger.js (hunk 2 above)
Comment (moniika) [reply core/block_dragger.js]: Updated

**Pull request merged**